pandas/tests/indexes/datetimes/test_date_range.py

```diff
@@ -88,6 +88,33 @@ def test_date_range_nat(self):
         with pytest.raises(ValueError, match=msg):
             date_range(start=pd.NaT, end='2016-01-01', freq='D')
 
+    def test_date_range_multiplication_overflow(self):
```

Comment:
so 2 test cases *fully* exercise all of this code???

Reply:
Good call, I'm tracking down untested cases now.